e2e/app/component/dialog.ts

```diff
@@ -1,57 +1,66 @@
+import Container from 'app/container';
+import {xPathOptionToXpath} from 'app/element/xpath-defaults';
 import {ElementHandle, Page} from 'puppeteer';
+import {ElementType} from '../xpath-options';
 
 export enum ButtonLabel {
   Confirm = 'Confirm',
   KeepEditing = 'Keep Editing',
+  Cancel = 'Cancel',
+  Calculate = 'Calculate',
+  AddThis = 'ADD THIS',
+  Finish = 'Finish',
 }
 
-const SELECTOR = {
-  dialogRoot: '.ReactModal__Content[role="dialog"]',
-  button: '[role="button"]'
+const Selector = {
+  defaultDialog: '//*[@role="dialog"]',
 }
 
-export default class Dialog {
-  private dialogElement: ElementHandle;
+export default class Dialog extends Container {
 
-  constructor(private readonly page: Page) {
+  private dialogElement: ElementHandle;
 
+  constructor(page: Page, xpath: string = Selector.defaultDialog) {
```

Comment:
One thing I'm not 100% sure about is the protocol here. You instantiate the Dialog object before you've determined that a dialog is ready or exists at all. I kinda feel like the `Dialog` should be something you wait on, and then it exposes properties that dialogs have.

Reply:
Sorry, I'm confused by this comment. Can you elaborate? Class constructor does nothing else except instantiate `page` and `xpath` class fields. It is not yet finding dialog. No action is happening here.